src/plugins/map/MQ2MapCommands.cpp

```diff
@@ -664,17 +672,20 @@ void MapSetLocationCmd(SPAWNINFO* pChar, char* szLine)
 		// FIXME: exception handling?
 		CVector3 pos{ std::stof(xloc), std::stof(yloc), std::stof(zloc) };
 
-		MakeMapLoc(params, label, tag, pos, isDefaultLocSettings);
+		MapLocTemplate* newMapLoc = new MapLocTemplate(params, label, tag, pos, isDefaultLocSettings);
+
+		gMapLocTemplates.push_back(newMapLoc);
+
+		UpdateMapLocIndexes();
+
+		std::string_view labelStr = gMapLocTemplates[gMapLocTemplates.size() - 1]->GetMapLoc()->GetLabelText();
```

Comment:
you can use `newMapLoc` instead of `gMapLocTemplates[gMapLocTemplates.size() - 1]`